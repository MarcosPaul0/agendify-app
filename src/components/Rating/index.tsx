import { View } from 'react-native';
import { Star } from 'phosphor-react-native';
import { COLORS } from '@constants/colors.constant';
import { IRatingProps } from './interfaces/ratingProps.interface';

export function Rating({ name, rating }: IRatingProps) {
  return (
    <View className="flex-row py-1">
      {Array.from({
        length: 5,
      }).map((_, index) => {
        const starIsFilled = index < rating ? 'fill' : 'regular';
        const key = `${name}-${index}`;

        return (
          <Star
            key={key}
            size={16}
            weight={starIsFilled}
            color={COLORS.WARNING}
          />
        );
      })}
    </View>
  );
}
