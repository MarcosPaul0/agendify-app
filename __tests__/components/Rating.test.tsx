import { render } from '@testing-library/react-native';
import { Rating } from '@components/Rating';

describe('Rating', () => {
  it('Match with snapshot', async () => {
    const tree = render(<Rating name="fake-name" rating={5} />);

    expect(tree).toMatchSnapshot();
  });
});
