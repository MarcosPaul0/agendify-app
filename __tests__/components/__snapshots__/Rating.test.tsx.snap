// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating Match with snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "paddingBottom": 4,
        "paddingTop": 4,
      },
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="16"
    bbWidth="16"
    color="#FFD015"
    fill="#FFD015"
    focusable={false}
    height={16}
    meetOrSlice={0}
    minX={0}
    minY={0}
    size={16}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          undefined,
          undefined,
          {},
        ],
        {
          "flex": 0,
          "height": 16,
          "width": 16,
        },
      ]
    }
    tintColor="#FFD015"
    vbHeight={256}
    vbWidth={256}
    width={16}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4294955029,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGRect
        fill={null}
        height="256"
        propList={
          [
            "fill",
          ]
        }
        width="256"
        x="0"
        y="0"
      />
      <RNSVGPath
        d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="16"
    bbWidth="16"
    color="#FFD015"
    fill="#FFD015"
    focusable={false}
    height={16}
    meetOrSlice={0}
    minX={0}
    minY={0}
    size={16}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          undefined,
          undefined,
          {},
        ],
        {
          "flex": 0,
          "height": 16,
          "width": 16,
        },
      ]
    }
    tintColor="#FFD015"
    vbHeight={256}
    vbWidth={256}
    width={16}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4294955029,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGRect
        fill={null}
        height="256"
        propList={
          [
            "fill",
          ]
        }
        width="256"
        x="0"
        y="0"
      />
      <RNSVGPath
        d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="16"
    bbWidth="16"
    color="#FFD015"
    fill="#FFD015"
    focusable={false}
    height={16}
    meetOrSlice={0}
    minX={0}
    minY={0}
    size={16}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          undefined,
          undefined,
          {},
        ],
        {
          "flex": 0,
          "height": 16,
          "width": 16,
        },
      ]
    }
    tintColor="#FFD015"
    vbHeight={256}
    vbWidth={256}
    width={16}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4294955029,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGRect
        fill={null}
        height="256"
        propList={
          [
            "fill",
          ]
        }
        width="256"
        x="0"
        y="0"
      />
      <RNSVGPath
        d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="16"
    bbWidth="16"
    color="#FFD015"
    fill="#FFD015"
    focusable={false}
    height={16}
    meetOrSlice={0}
    minX={0}
    minY={0}
    size={16}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          undefined,
          undefined,
          {},
        ],
        {
          "flex": 0,
          "height": 16,
          "width": 16,
        },
      ]
    }
    tintColor="#FFD015"
    vbHeight={256}
    vbWidth={256}
    width={16}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4294955029,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGRect
        fill={null}
        height="256"
        propList={
          [
            "fill",
          ]
        }
        width="256"
        x="0"
        y="0"
      />
      <RNSVGPath
        d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="16"
    bbWidth="16"
    color="#FFD015"
    fill="#FFD015"
    focusable={false}
    height={16}
    meetOrSlice={0}
    minX={0}
    minY={0}
    size={16}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          undefined,
          undefined,
          {},
        ],
        {
          "flex": 0,
          "height": 16,
          "width": 16,
        },
      ]
    }
    tintColor="#FFD015"
    vbHeight={256}
    vbWidth={256}
    width={16}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4294955029,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGRect
        fill={null}
        height="256"
        propList={
          [
            "fill",
          ]
        }
        width="256"
        x="0"
        y="0"
      />
      <RNSVGPath
        d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
